hudi-cli/src/main/java/org/apache/hudi/cli/commands/ExportCommand.java

```diff
@@ -0,0 +1,230 @@
+
+    if (descending) {
+      Collections.reverse(nonArchivedInstants);
+      numCopied = copyNonArchivedInstants(nonArchivedInstants, numExports, localFolder);
+      if (numCopied < numExports) {
+        Collections.reverse(archivedStatuses);
+        numCopied += copyArchivedInstants(archivedStatuses, actionSet, numExports - numCopied, localFolder);
+      }
+    } else {
+      numCopied = copyArchivedInstants(archivedStatuses, actionSet, numExports, localFolder);
+      if (numCopied < numExports) {
+        numCopied += copyNonArchivedInstants(nonArchivedInstants, numExports - numCopied, localFolder);
+      }
+    }
+
+    return "Exported " + numCopied + " Instants to " + localFolder;
+  }
+
+  private int copyArchivedInstants(List<FileStatus> statuses, Set<String> actionSet, int limit, String localFolder) throws Exception {
+    int copyCount = 0;
+
+    for (FileStatus fs : statuses) {
+      // read the archived file
+      Reader reader = HoodieLogFormat.newReader(FSUtils.getFs(HoodieCLI.getTableMetaClient().getBasePath(), HoodieCLI.conf),
+          new HoodieLogFile(fs.getPath()), HoodieArchivedMetaEntry.getClassSchema());
+
+      // read the avro blocks
+      while (reader.hasNext() && copyCount < limit) {
+        HoodieAvroDataBlock blk = (HoodieAvroDataBlock) reader.next();
+        for (IndexedRecord ir : blk.getRecords()) {
+          // Archived instants are saved as arvo encoded HoodieArchivedMetaEntry records. We need to get the
+          // metadata record from the entry and convert it to json.
+          HoodieArchivedMetaEntry archiveEntryRecord = (HoodieArchivedMetaEntry) SpecificData.get()
+              .deepCopy(HoodieArchivedMetaEntry.SCHEMA$, ir);
+
+          final String action = archiveEntryRecord.get("actionType").toString();
+          if (!actionSet.contains(action)) {
+            continue;
+          }
+
+          GenericRecord metadata = null;
+          switch (action) {
+            case HoodieTimeline.CLEAN_ACTION:
+              metadata = archiveEntryRecord.getHoodieCleanMetadata();
+              break;
+            case HoodieTimeline.COMMIT_ACTION:
+            case HoodieTimeline.DELTA_COMMIT_ACTION:
+              metadata = archiveEntryRecord.getHoodieCommitMetadata();
+              break;
+            case HoodieTimeline.ROLLBACK_ACTION:
+              metadata = archiveEntryRecord.getHoodieRollbackMetadata();
+              break;
+            case HoodieTimeline.SAVEPOINT_ACTION:
+              metadata = archiveEntryRecord.getHoodieSavePointMetadata();
+              break;
+            case HoodieTimeline.COMPACTION_ACTION:
+              metadata = archiveEntryRecord.getHoodieCompactionMetadata();
+              break;
+            default:
+              throw new RuntimeException("Unknown type of action " + action);
```

Comment:
Hi @prashantwason, we can use `HoodieException`

Reply:
Done